daq/runner.py

```diff
@@ -870,6 +870,25 @@ def _combine_result_set(self, set_key, result_sets):
                 results.append('%s:%s:%s' % (set_key, name, status))
         return results
 
+    def _send_device_result(self, mac, test_results):
+        if not self._device_result_client:
+            return
+
+        if test_results is None:
+            device_result = PortBehavior.failed
+        else:
+            device_result = self._calculate_device_result(test_results)
+
+        LOGGER.info('Sending device result for device %s: %s', mac, device_result)
+        self._device_result_client.send_device_result(mac, device_result)
+
+    def _calculate_device_result(self, test_results):
+        for module_name, module_result in test_results.get('modules', {}).items():
+            for test_name, test_result in module_result.get('tests', {}).items():
```

Comment:
@grafnu  @didovesei I think this is right, but just flagging that this is different from the current report generation which compares against a 'required_result' from the module_config to determine if a test passes/fails. ( https://github.com/faucetsdn/daq/blob/34d14e92e4329138ca213de6f495ae3713b1b1c0/daq/report.py#L263  ) I can't find an instance where a test is expected to produce a "fail" result and looks like it was only in place for the info tests (which are being converted into normal pass/fail tests)